freeipa/src/main/java/com/sequenceiq/freeipa/sync/saltstatus/StackSaltStatusCheckerJob.java

```diff
@@ -0,0 +1,109 @@
+package com.sequenceiq.freeipa.sync.saltstatus;
+
+import static com.sequenceiq.cloudbreak.util.Benchmark.checkedMeasure;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.quartz.statuschecker.job.StatusCheckerJob;
+import com.sequenceiq.flow.core.FlowLogService;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.freeipa.salt.rotatepassword.event.RotateSaltPasswordReason;
+import com.sequenceiq.freeipa.orchestrator.RotateSaltPasswordService;
+import com.sequenceiq.freeipa.service.GatewayConfigService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+import com.sequenceiq.freeipa.sync.InterruptSyncingException;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class StackSaltStatusCheckerJob extends StatusCheckerJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackSaltStatusCheckerJob.class);
+
+    @Inject
+    private StackSaltStatusCheckerJobService jobService;
+
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private RotateSaltPasswordService rotateSaltPasswordService;
+
+    public StackSaltStatusCheckerJob(Tracer tracer) {
+        super(tracer, "Stack Salt Status Checker Job");
+    }
+
+    @Override
+    protected Object getMdcContextObject() {
+        return stackService.getStackById(getStackId());
+    }
+
+    private Long getStackId() {
+        return Long.valueOf(getLocalId());
+    }
+
+    @Override
+    protected void executeTracedJob(JobExecutionContext context) throws JobExecutionException {
+        Long stackId = getStackId();
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        MDCBuilder.buildMdcContext(stack);
+        try {
+            rotateSaltPasswordService.validateRotateSaltPassword(stack);
+        } catch (BadRequestException e) {
+            LOGGER.info("StackSaltStatusCheckerJob cannot run, because validation failed for stack {} with message: {}", stackId, e.getMessage());
+            jobService.unschedule(context.getJobDetail().getKey());
+        }
+        try {
```

Comment:
I assume this code block should run only when the previous validation is passed e.g. when the job cannot run, this code should be skipped.